integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataPageSource.java

```diff
@@ -166,61 +146,32 @@ protected void closeWithSuppression(Throwable throwable)
   /**
    * Lazy Block Implementation for the Carbondata
    */
-  private final class CarbondataBlockLoader
-      implements LazyBlockLoader<LazyBlock>
-  {
+  private final class CarbondataBlockLoader implements LazyBlockLoader<LazyBlock> {
     private final int expectedBatchId = batchId;
     private final int columnIndex;
-    private final Type type;
     private boolean loaded;
 
-    public CarbondataBlockLoader(int columnIndex, Type type)
-    {
+    CarbondataBlockLoader(int columnIndex) {
       this.columnIndex = columnIndex;
-      this.type = requireNonNull(type, "type is null");
     }
 
-    @Override
-    public final void load(LazyBlock lazyBlock)
-    {
+    @Override public final void load(LazyBlock lazyBlock) {
       if (loaded) {
         return;
       }
       checkState(batchId == expectedBatchId);
       try {
-        Block block = readers[columnIndex].readBlock(type);
+        PrestoVectorBlockBuilder blockBuilder =
```

Comment:
getting presto block from the update streamReaders which extends CarbonColumnVectorImpl and implements PrestoVectorBlockBuilder